### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by johnworth**
Splits up the one large transaction into multiple, shorter-lived transactions per job.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -234,24 +222,29 @@ func (p *Propagator) JobUpdates(extID string, maxRetries int64) ([]DBJobStatusUp
...
-	}
 	return retval, err
 }
 
 // MarkPropagated marks the job as propagated in the database as part of the
 // transaction tracked by the *Propagator.
 func (p *Propagator) MarkPropagated(id string) error {
+	tx, err := p.db.Begin()
+	if err != nil {
+		tx.Rollback()
```
Comment (johnworth) [inline main.go]: I'm iffy on this Rollback() call.

File: main.go [hunk 2]
```diff
@@ -272,12 +265,18 @@ func (p *Propagator) StorePropagationAttempts(update *DBJobStatusUpdate) error {
...
-	insertStr := `UPDATE ONLY job_status_updates
-		         SET propagation_attempts = $2,
-		             last_propagation_attempt = $3
-		       WHERE id = $1`
-	_, err := p.tx.Exec(insertStr, id, newVal, lastAttemptTime)
-	return err
+	insertStr := `UPDATE ONLY job_status_updates SET propagation_attempts = $2, last_propagation_attempt = $3 WHERE id = $1`
+	tx, err := p.db.Begin()
+	if err != nil {
+		tx.Rollback()
```
Comment (johnworth) [inline main.go]: Another Rollback() call that I'm iffy on.

File: main.go (hunk 1 above)
Comment (slr71) [reply main.go]: I'm relatively certain that `tx` will be  nil if an error occurs in `Begin`. You can omit these rollbacks.

**Review by slr71 — APPROVED** [review]
The changes look good to me and the lint error is a little bizarre. :shipit:

**Pull request merged**